app/controllers/stats_controller.rb

```diff
@@ -82,6 +82,34 @@ def self.closed_cases_correspondence_type
 
   private
 
+  def generate_final_report(report, report_data=nil)
```

Comment:
This could possibly be a case statement with smaller methods for each case - might make it easier to parse, I'm thinking something like:

```ruby
def generate_final_report(report, report_data = nil)
    case report.report_format
    when 'xlsx'
       send_xlsx_report(report)
    when 'csv'
      send_csv_report(report)
    else
      send_default_report(report)
    end
end
```
not sure about the method names, but thinking that something like:

`genereate_xlsx_report`, `format_xlsx_report` or `send_xlsx_report` might be candidates.